### PR TITLE
Improve dev setup experience

### PR DESCRIPTION
**Pull request by benarmston**
Add a `.env.development` file which configures the app to work correctly on the `openflight-vagrant-cluster`.  By default it uses the production API services.

### CODE REVIEW

**Comment by benarmston** [comment]
@WilliamMcCumstie do `.env.development` and `DEVELOPMENT.md` make sense to you?  Do they seem like they'd be easy to follow?  (I've still got a few commits to push to openflight-development-cluster, so that's not quite working yet.).

File: .env.development [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+REACT_APP_LOGIN_API_BASE_URL="/login/api/v0"
+REACT_APP_API_BASE_URL="/desktop/api/v2"
+REACT_APP_CONFIG_FILE="/dev/desktop/config.prod-services.json"
+# ==============================================================================
+
+# ==============================================================================
+# ALTERNATE configure paths to external services.
+#
+# If you are not using `openflight-vagrant-cluster` you can configure
+# `create-react-app` to proxy to the services for you.
```
Comment (WilliamMcCumstie) [inline .env.development:31]: What is `create-react-app`?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+```
+cd /code/flight-desktop-webapp
+/opt/flight/bin/yarn run start 
+```
+
+## Accessing the development build
+
+Once your OpenFlight Vagrant cluster is created, the development build of
+Flight Desktop Webapp will be available at
+https://flight.lvh.me:8443/dev/desktop.
```
Comment (WilliamMcCumstie) [inline DEVELOPMENT.md:36]: AFAICS it starts on port `3001` not port `8443`

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Flight Desktop Webapp is designed to be developed on an [OpenFlight Vagrant
+Cluster](https://github.com/openflighthpc/openflight-vagrant-cluster).  The
+OpenFlight Vagrant Cluster repository contains details of how to install,
+configure and build an OpenFlightHPC cluster using Vagrant.  Follow the
+instructions there and then continue with the instructions here.
+
+## Configuring which services Flight Desktop Webapp uses
+
+By default, Flight Desktop Webapp is configured to use the production services
+installed on your OpenFlight Vagrant cluster.  You can configure the services
```
Comment (WilliamMcCumstie) [inline DEVELOPMENT.md:18]: Why are they tied to the production  services on the `OpenFlight Vagrant Cluster`? Would they not work on any machine with the production builds?

Also which services are required? What if I am having issues with a dev build of `flight-www`, will this affect the reverse proxy?

File: public/config.prod-services.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "apiRootUrl": "/desktop/api/v2",
+  "websocketPathPrefix": "/ws",
+  "websocketPathIp": "127.0.0.1"
```
Comment (WilliamMcCumstie) [inline public/config.prod-services.json:4]: There is still no easy way to change the `"devOnlyWebsocketRootUrl`. Everything seems to support proxy-ing except the websocket connection.

Would this be better suited as an env var?

**Review by WilliamMcCumstie — COMMENTED** [review]
In principle the changes look okay, made a few comments about various things.

I see we are heavily tying ourselves to `openflight-vagrant-cluster` which is a new project I haven't used much. It also has a few bugs in it which prevent me from testing it out.

My one major concern is the documentation is very `ItJustWorks` :tm:. We are relying on these external services to be function correctly, but with very little information how. If anything ever went wrong, it gives few details about how to go about fixing it.

Having an external vagrant build provide the packages is fine, however the development guide should focus on:

* How to be configured with `production` builds not the vagrant cluster, and
* Which rpms provide the services and their minimum tested version.

I don't believe `openflight-vagrant-cluster` provides any services which can not be obtained through `rpm`. There are a few mentions of "the reverse proxy", however assume you are referring to `nginx`/`flight-www`? By calling it the `openflight-vagrant-cluster` reverse proxy it sounds like something custom to the dev environment.

File: .env.development [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# See the `.env` file for comments.
+PORT=3001
+PUBLIC_URL="/dev/desktop"
```
[the remark this answers is not in the document]
Comment (WilliamMcCumstie) [reply .env.development:3]: Discussed IRL

File: DEVELOPMENT.md (hunk 2 above)
Comment (WilliamMcCumstie) [reply DEVELOPMENT.md]: Discussed IRL

**Pull request merged**